crates/teloxide/src/update_listeners/polling.rs

```diff
@@ -436,10 +438,23 @@ impl<B: Requester> Stream for PollingStream<'_, B> {
                     }
                 }
                 Err(err) => {
-                    // Prevents the CPU spike occuring at network connection lose: <https://github.com/teloxide/teloxide/issues/780>
-                    let backoff_strategy = &this.polling.backoff_strategy;
-                    this.eepy.set(Some(sleep(backoff_strategy(*this.error_count))));
-                    log::trace!("set {:?} reconnection delay", backoff_strategy(*this.error_count));
+                    /*
+                       In case of RetryAfter(..) error we pause the polling for the specified amount
+                       of seconds.
+
+                       Otherwise, in case of network connection lose (<https://github.com/teloxide/teloxide/issues/780>) we
+                       use the backoff strategy to prevent the high CPU usage due to multiple instant reconnections
+                    */
+                    let delay = match err.retry_after() {
+                        Some(seconds) => {
+                            *this.error_count = 0;
+                            seconds.duration()
+                        }
+                        None => (this.polling.backoff_strategy)(*this.error_count),
```

Comment:
we should increment `error_count` here, instead of when `eepy` compiled, so that `= 0` in the other branch works as intended. 

not sure if we increment before it after getting backoff strategy (kind of confusing both ways tbh...)

Reply:
Increment after getting backoff strategy is okay, this way we get the proper backoff timing sequence (start from 1s instead of 2s)